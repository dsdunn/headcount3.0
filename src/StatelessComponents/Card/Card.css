.card {
  height: 300px;
  width: 150px;
  border: 1px solid blue;
}

.district {
  height: 20px;
}

.card ul {
  list-style: none;
  padding-left: 15%;
  padding-right: 15%;
}

.card .year {
  width: 50%;
  text-align: right;
  margin: auto 8px;
}

.card .percent {
  width: 50%;
  text-align: left;
  margin: auto 8px;
}

.district {
  text-align: center;
}

.low {
  background-color: #f4aca8;
}