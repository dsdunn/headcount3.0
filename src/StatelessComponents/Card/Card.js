import React from 'react';

const Card = ({ district }) => {

  const generateList = (stats) => {
    const years = Object.keys(stats);
    years.map(year => {
      return (
        <li>{year}<span className="percent">{stats[year]}</span></li>
        )
    })

  }

  return(
    <div>
      <h5 className="district">Some District</h5>
      <ul>
        {() => generateList(district.stats)}
      </ul>
    </div>
    )
}

export default Card;