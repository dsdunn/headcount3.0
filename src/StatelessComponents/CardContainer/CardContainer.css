.card-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  border-top: 2px solid #008A92;
  padding: 3%;
}