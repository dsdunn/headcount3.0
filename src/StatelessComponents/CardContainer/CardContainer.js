import React from 'react';
import Card from '../Card/Card';

const CardContainer = () => {
  return (
    <div>
      <h5> this is the card container </h5>
    </div>
    )
}

export default CardContainer;