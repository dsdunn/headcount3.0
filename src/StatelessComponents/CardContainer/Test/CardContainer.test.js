import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import CardContainer from '../CardContainer';
import Card from '../../Card/Card';

describe('CardContainer', () => {
  let wrapper = shallow(
    <CardContainer 
      allDistricts={[{ location: 'COLORADO', stats: { 2007: 0.111 } }]}
      selectedCards={[{ location: 'COLORADO', stats: { 2007: 0.111 } }]}
      selectDistrict={jest.fn()}
      unselect={jest.fn()}
    />);
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should assign the className="selected" to cards of selected districts', () => {
    wrapper = mount(
      <CardContainer
        allDistricts={[
          { location: 'COLORADO', stats: { 2007: 0.111 } }, 
          { location: 'COLORADO 20', stats: { 2007: 0.111 } }, 
          { location: 'ARAPAHO', stats: { 2007: 0.111 } }, 
          { location: 'PINE NUTS', stats: { 2007: 0.111 } }]}
        selectDistrict={jest.fn()}
        selectedCards={[
          { location: 'COLORADO', stats: { 2007: 0.111 } }, 
          { location: 'COLORADO 20', stats: { 2007: 0.111 } }]}
        unselect={jest.fn()}
      />);

    expect(wrapper.find('.selected').length).toEqual(2);
  });
});