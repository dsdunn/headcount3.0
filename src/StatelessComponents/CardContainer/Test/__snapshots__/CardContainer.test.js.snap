// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    allDistricts={
        Array [
            Object {
              "location": "COLORADO",
              "stats": Object {
                "2007": 0.111,
              },
            },
          ]
    }
    selectDistrict={[Function]}
    selectedCards={
        Array [
            Object {
              "location": "COLORADO",
              "stats": Object {
                "2007": 0.111,
              },
            },
          ]
    }
    unselect={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          district={
                    Object {
                              "location": "COLORADO",
                              "stats": Object {
                                "2007": 0.111,
                              },
                            }
          }
          selectDistrict={[Function]}
          selected={true}
          unselect={[Function]}
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "COLORADO",
        "nodeType": "function",
        "props": Object {
          "district": Object {
            "location": "COLORADO",
            "stats": Object {
              "2007": 0.111,
            },
          },
          "selectDistrict": [Function],
          "selected": true,
          "unselect": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            district={
                        Object {
                                    "location": "COLORADO",
                                    "stats": Object {
                                      "2007": 0.111,
                                    },
                                  }
            }
            selectDistrict={[Function]}
            selected={true}
            unselect={[Function]}
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "COLORADO",
          "nodeType": "function",
          "props": Object {
            "district": Object {
              "location": "COLORADO",
              "stats": Object {
                "2007": 0.111,
              },
            },
            "selectDistrict": [Function],
            "selected": true,
            "unselect": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
